### PR TITLE
[air] Add initial benchmark section

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

Initial benchmarks for XGBoost. Pytorch benchmarks to be added later.

Scripts and config will be added once merged in.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]
Looks great. A few comments on the script layout (can be addressed separately):
- Could we have separate scripts for training and prediction? This is for readability: it's not great to have multiple workflows in the same script.
- Should make `@run_and_time_it` a common util
- remove comment for `# silver tier`
- `to_air_checkpoint(".", model)` <--- isn't the "." fishy?
- `batch_predictor.predict(ds.drop_columns(["labels"]))` <--- I think you can use the "feature_columns" arg of predict() that Amog added, instead of dropping labels.

File: doc/source/ray-air/benchmarks.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+- `XGBoost Training Script`_
+- `XGBoost Cluster configuration`_
+
+.. list-table::
+
+    * - **Cluster Setup**
+      - **# workers**
+      - **Data Size**
+      - **# of rows**
+      - **Time taken**
```
Comment (ericl) [inline doc/source/ray-air/benchmarks.rst:59]: Add **Command** column.

(e.g., `python xgboost_predict.py --rows=1m`)

**Review by ericl — CHANGES_REQUESTED** [review]
One request to add the exact reproduction command.

File: doc/source/ray-air/benchmarks.rst [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      - 10 GB
+      - 26M rows
+      - 692 s
+      - `python xgboost_benchmark.py --size 10GB`
+    * - 10 m5.4xlarge nodes
+      - 10 actors (12 CPUs each)
+      - 100 GB
+      - 260M rows
+      - 693 s
+      - `python xgboost_benchmark.py --size 100GB`
```
Comment (ericl) [inline doc/source/ray-air/benchmarks.rst:75]: Does this actually work as written? Shouldn't num_actors be here somewhere?

**Review by ericl — APPROVED** [review]
LGTM

**Comment by richardliaw** [comment]
will merge when lint passes then
